src/server/bg_services/replication_log_parser.js

```diff
@@ -66,6 +76,79 @@ async function get_aws_log_candidates(source_bucket_id, rule_id, replication_con
     };
 }
 
+async function get_azure_log_candidates(source_bucket_id, rule_id, replication_config, candidates_limit, sync_deletions) {
+    const source_bucket = system_store.data.get_by_id(source_bucket_id);
+    const namespace_resource = source_bucket.namespace.write_resource.resource;
+    const src_storage_account = namespace_resource.connection.access_key;
+    const src_container_name = namespace_resource.connection.target_bucket;
+    const prefix = replication_config.log_replication_info.azure_log_replication_info.prefix || '';
+    const { logs_query_client, monitor_workspace_id } = get_source_bucket_azure_connection(source_bucket_id);
+    let candidates;
+
+    // Please note that we don't use "where Time > ago" - the time is passed as "duration" in the query options
+    const kusto_query =
+    `StorageBlobLogs
+    | project Time=TimeGenerated, Action=substring(Category, 7), Key=ObjectKey
+    | sort by Time asc
+    | where Action == "Write" or Action == "Delete"
+    | where Key startswith "/${src_storage_account.unwrap()}/${src_container_name}/${prefix}"
+    | where Key !contains "test-delete-non-existing-"
+    | parse Key with * "/" StorageAccount "/" Container "/" Key
+    | project Time, Action, Key`;
+
+    //Our "continuation token" is the timestamp of the the last log retrieval
+    let continuation_token = get_continuation_token_for_rule(rule_id, replication_config);
+    let duration;
+    if (continuation_token) {
+        duration = `${(Date.now() - parseInt(continuation_token, 10))}ms`;
+    } else {
+        duration = Durations.oneHour;
+    }
+    continuation_token = Date.now();
+    const query_result = await logs_query_client.queryWorkspace(
+    monitor_workspace_id.unwrap(),
+    kusto_query,
+    // The timespan is an ISO8601 formatted time (or interval). Some common aliases
+    // are available (like OneDay, OneHour, FoutyEightHours, etc..) but any properly formatted ISO8601
+    // value is valid.
+    // If a continuation token does not yet exist, we'll take all logs from the past hour
+    { duration: duration },
```

Comment:
What's the size of the runtime object response of one-hour logs (specifically in rush hour)? I am afraid it'll be too big and we will get oom killed

Reply:
what I'm trying to say is that we need to check if one hour can get us a very big object which we will not be able to handle/crash

Reply:
I agree. We need to test and tune it. We can do it after this PR